### ISSUE TITLE
CTC -  No valid path found

### ISSUE THREAD

**Issue by akshatdewan** [rwth-i6/returnn-experiments#1]
Hi, 

I am trying to reproduce your results for Librispeech dataset (train-clean-100+ train-clean-360+train-other-500). I follow all the step scripts here https://github.com/rwth-i6/returnn-experiments/tree/master/2018-asr-attention/librispeech/full-setup-attention before running the 22_train.sh and they all seem to run fine without errors. 

But I am getting 
`2018-07-17 16:19:35.497434: W tensorflow/core/util/ctc/ctc_loss_calculator.cc:144] No valid path found.
`
messages every few batches. Should I worry about it? Or there is something going wrong with my training? https://stackoverflow.com/questions/45130184/ctc-loss-error-no-valid-path-found suggests that it is caused when the length of the label sequence is greater than the length of the input sequence. I am filtering out audio sequences shorter than 1 sec and larger than 15 secs. 

Thanks
Best!
Akshat

**Comment by albertz** [comment]
Yes, that happens when the target seq len is longer than the source len (which is the case initially for some cases because of a high reduction factor). If it happens every single step, something is wrong, but if it happens occasionally, this is fine. Can you just continue running the training and see if it works? In later epochs, it should not happen anymore (or much rarely).

Best,
Albert

**Comment by akshatdewan** [comment]
Hi Albert, 

Many thanks for the explanation. It makes sense now. I ran the training and it finished 32 epochs of pretraining and the subject mentioned messages are much more rare now. I am actually training on a dataset that is 2 times the size of librispeech (librispeech + proprietary data). I tested one of the models (after 29 epochs) by running 23_recog.sh and I get meaningful hypotheses. I will let it run for 100 epochs and see how it goes. 

Thanks for making this toolkit available and your support.
Akshat
